dotCMS/src/main/java/com/dotmarketing/servlets/InitServlet.java

```diff
@@ -196,7 +197,17 @@ public void init(ServletConfig config) throws ServletException {
         }
 
         //Initialize the Cached Vanity URL cache
-        VanityUrlServices.getInstance().initializeVanityUrlCache();
+        try {
+            VanityUrlServices.getInstance().initializeVanityUrlCache();
+        } catch (Exception e) {
+            if (e instanceof IndexMissingException || e.getCause() instanceof IndexMissingException) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1193)